extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -838,6 +857,354 @@ public void run()
     return success();
   }
 
+  private TaskStatus runLegacy(final TaskToolbox toolbox) throws Exception
```

Comment:
Yikes that's a lot of code duplication. IF we get rid of this ASAP as soon as 0.11.1 is cut then I'm okay with this, but that would mean that there would be no upgrade path from 0.11.0 to 0.12 or whatever the next version is which I don't think is a great restriction either. If we don't want to have this restriction, then the legacy logic may need to stick around for a bit longer which would make this duplication pretty ugly. Any thoughts? Maybe @gianm can weigh in too.